subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ResolvedFilesCollectingVisitor.java

```diff
@@ -70,11 +74,7 @@ public void visitFile(ComponentArtifactIdentifier artifactIdentifier, AttributeC
     private void addArtifactFiles() {
         if (!artifacts.isEmpty()) {
             for (ResolvedArtifact artifact : artifacts) {
-                try {
-                    this.files.add(artifact.getFile());
-                } catch (Throwable t) {
-                    failures.add(t);
-                }
+                this.files.add(artifact.getFile());
```

Comment:
⭕️ It seems like this class never actually uses the `artifacts` themselves, so it would be simpler to read if we colltected an `artifactFiles` collection instead. Might also be more efficient since we then don't call `getFile()` twice on each artifact.